syft/pkg/cataloger/common/cpe/generate_test.go

```diff
@@ -768,6 +768,14 @@ func TestCandidateProducts(t *testing.T) {
 			},
 			expected: []string{"spring_framework", "springsource_spring_framework" /* <-- known good names | default guess --> */, "springframework"},
 		},
+		{
+			name: "spring-security-core",
+			p: pkg.Package{
+				Name: "spring-security-core",
+				Type: pkg.JavaPkg,
+			},
+			expected: []string{"spring-security-core", "spring_security", "spring_security_core"},
+		},
```

Comment:
I saw in the above additional vendors is also using `vmware`. 

Do we want to add a case here under `TestCandidateVendor` to cover `spring-security-core` as well?